app/helpers/application_helper.rb

```diff
@@ -286,4 +286,122 @@ def sign_in_path
+  # Returns the object in its current state if its alive else the state before or after this version (after if its a create event's version)
+  def get_version_object(version)
+    version.item || version.reify || version.next.reify
+  end
+
+  def event_change_description(version)
+    case
+    when version.event == 'create' then 'submitted new'
+
+    when version.event == 'destroy' then 'deleted'
+
+    when version.changeset['state']
+      case version.changeset['state'][1]
+      when 'unconfirmed' then 'accepted'
+      when 'withdrawn' then 'withdrew'
+      when 'canceled', 'rejected', 'confirmed' then version.changeset['state'][1]
+      when 'new' then 'resubmitted'
+      end
+
+    when version.changeset['start_time'] && version.changeset['start_time'][0].nil?
+      'scheduled'
+
+    when version.changeset['start_time'] && version.changeset['start_time'][1].nil?
+      'unscheduled'
+
+    else
+      "updated #{updated_attributes(version)} of"
+    end
+  end
+
+  def users_role_change_description(version)
+    version.event == 'create' ? 'added' : 'removed'
+  end
+
+  def subscription_change_description(version)
+    user = get_version_object(version).user
+    user_name = user.name unless user.id.to_s == version.whodunnit
+    version.event == 'create' ? "subscribed #{user_name} to" : "unsubscribed #{user_name} from"
+  end
+
+  def registration_change_description(version)
+    if version.item_type == 'Registration'
+      user = get_version_object(version).user
+    else
+      registration_id = get_version_object(version).registration_id
+      registration_last_version = PaperTrail::Version.where(item_type: 'Registration', item_id: registration_id).last
+      user = get_version_object(registration_last_version).user
+    end
+
+    if user.id.to_s == version.whodunnit
+      case version.event
+      when 'create' then 'registered to'
+      when 'update' then "updated #{updated_attributes(version)} of the registration for"
+      when 'destroy' then 'unregistered  from'
+      end
+    else
+      case version.event
+      when 'create' then "registered #{user.name} to"
+      when 'update' then "updated #{updated_attributes(version)} of  #{user.name}'s registration for"
+      when 'destroy' then "unregistered #{user.name} from"
+      end
+    end
+  end
+
+  def comment_change_description(version)
+    user = get_version_object(version).user
+    if user.id.to_s == version.whodunnit && version.event == 'create'
+      'commented on'
+    elsif user.id.to_s != version.whodunnit && version.event == 'create'
+      "re-added #{user.name}'s comment on"
+    else
+      "deleted #{user.name}'s comment on"
+    end
+  end
+
+  def vote_change_description(version)
+    user = get_version_object(version).user
+    if user.id.to_s == version.whodunnit && version.event == 'create'
+      'voted on'
+    elsif user.id.to_s != version.whodunnit && version.event == 'create'
+      "re-added #{user.name}'s vote on"
```

Comment:
I think it is best if we use the phrasing 're-added User's X vote' in both case. Because then we can tell the difference between a user going to the proposal and voting, and a user reverting a change from the changelog.